### PR TITLE
Safer dependencies collector

### PR DESCRIPTION
**Pull request by efratas**
some improvements such as using path.join,
fix scan source dev and peer dependencies were not analyzed

related to https://github.com/apiiro/combobulator/issues/21

### CODE REVIEW

File: src/registry/maven.py [hunk 1]
```diff
@@ -35,7 +36,7 @@ def recv_pkg_info(pkgs, url=REGISTRY_URL):
 
 def scan_source(dir):
     try:
-        path = dir + "./pom.xml"
+        path = os.path.join(dir, "./pom.xml")
```
Comment (moshe-apiiro) [inline src/registry/maven.py:39]: os is not imported

File: src/registry/npm.py [hunk 2]
```diff
@@ -48,20 +50,18 @@ def recv_pkg_info(pkgs, url=REGISTRY_URL):
             i.exists = False
             
 
-def scan_source(dir):
+def scan_source(dir_):
```
Comment (moshe-apiiro) [inline src/registry/npm.py:53]: why to introduce inconsistency such as dir_ here and dir elsewhere?(maven for example)

File: src/registry/maven.py (hunk 1 above)
Comment (efratas) [reply src/registry/maven.py:39]: first line, it is..

File: src/registry/npm.py (hunk 2 above)
Comment (efratas) [reply src/registry/npm.py:53]: You're correct, should have change all dir occurrences  (or rename to directory)

**Comment by efratas** [comment]
@elishapiiro i see what you did there..

**Review by Matan-Giladi — APPROVED** [review]

**Comment by efratas** [comment]
dead repository

**Pull request closed without merging**